src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,210 @@
+    * @return Array that specifies the x, y and z components of the joint plane
+    */
+  private def findJointNormal(strike: Double, dip: Double): Array[Double] = {
+    val strikeRadians = strike.toRadians
+    val dipRadians = dip.toRadians
+    val strikeVector = DenseVector[Double](math.cos(-strikeRadians), math.sin(-strikeRadians), 0.0)
+    // If joint is vertical, dip vector will point along negative z-axis
+    val dipVector = if (math.abs(dip - 90.0) < NumericUtils.EPSILON) {
+      DenseVector[Double](0.0, 0.0, -1.0)
+    } else {
+      DenseVector[Double](math.cos(dipRadians) * math.cos(-(strikeRadians + math.Pi / 2.0)),
+        math.cos(dipRadians) * math.sin(-(strikeRadians + math.Pi / 2.0)),
+        -math.sin(dipRadians))
+    }
+    val jointNormal = linalg.normalize(linalg.cross(strikeVector, dipVector))
+    Array(jointNormal(0), jointNormal(1), jointNormal(2))
+  }
+
+  /**
+    * Finds the faces that define the rock volume of interest
+    *
+    * @param globalOrigin Coordinates of global origin
+    * @param rockVolume Array of arrays describing each of the faces that define the boundaries of
+    *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+    *                       y and z coordinates of a point that lies within the plane containing the face.
+    *                       Strike and dip should be specified in degrees.
+    * @return A sequence of faces that represent the rock volume of interest
+    */
+  private def findBoundingFaces(globalOrigin: Array[Double], rockVolume: Seq[InputFace]): Seq[Face] = {
+    rockVolume map { inputFace =>
+      val faceCenter = inputFace.pointInPlane
+      val normal = findJointNormal(inputFace.strike, inputFace.dip)
+      val distance = Joint.findDistance(normal, globalOrigin, faceCenter)
+      if (distance > 0.0) {
+        Face(normal, distance, inputFace.phi, inputFace.cohesion)
+      } else {
+        // Ensures that all normals are outward pointing
+        Face(Array(-normal(0), -normal(1), -normal(2)), -distance, inputFace.phi, inputFace.cohesion)
+      }
+    }
+  }
+
+  /**
+    * Finds representative joint for each joint set, centered at the lower left corner of the
+    * bounding box. For deterministic inputs, representative joint will have normal vector
+    * corresponding to strike and dip input by user as well as the specified phi and cohesion.
+    * Stochastic inputs are presently not supported, but will be implemented at a later time.
+    *
+    * @param globalOrigin Coordinates of the global origin
+    * @param lowerLeftCorner Lower left corner of the bounding box
+    * @param jointSets Array containing the input data representing joint sets
+    * @return Seq of "master" joints that represent input joint sets. These will be used as a starting point
+    *         to generate full joint set.
+    */
+  private def findMasterJoints(globalOrigin: Array[Double], lowerLeftCorner: Array[Double],
```

Comment:
Here's another spot where you could do an `assert` on the length of the `globalCoordinates` argument. I think there's more below, but I'll stop pointing them out now.